data/snippet/signed_request.toml

```diff
@@ -0,0 +1,49 @@
+id = "signed_request"
+name = "Signed Request/Response"
+description = "Check signatures of requestst and sign responses with a private key"
+code = """
+const crypto = require('crypto')
+
+const secret = 'SECRET_KEY'
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+
+async function createHexSignature(requestBody) {
+  let hmac = crypto.createHmac('sha256', secret)
+  hmac.update(requestBody)
+  return hmac.digest('hex')
+}
+
+async function checkSignature(request) {
+  // hash request with secret key
+  let expectedSignature = await createHexSignature(await request.text())
+  let actualSignature = await request.headers.get('signature')
+
```

Comment:
can't have \n\n. Hugo won't compile. I suggest running `npm run start-hugo` before submitted to check out what it looks like